druid/src/widget/container.rs

```diff
@@ -12,54 +12,59 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! A convenience widget that combines common styling and positioning widgets.
+//! Draws a border and a background around a widget or a widget tree.
 
-use crate::shell::kurbo::{Point, Rect, Size};
+use crate::shell::kurbo::{Point, Rect, RoundedRect, Size};
 use crate::{
     BoxConstraints, Data, Env, Event, EventCtx, LayoutCtx, PaintBrush, PaintCtx, RenderContext,
     UpdateCtx, Widget, WidgetPod,
 };
 
-struct BorderState {
+struct BorderStyle {
     width: f64,
     brush: PaintBrush,
 }
 
-#[derive(Default)]
-struct ContainerStyle {
+/// Draws a border and a background around a widget or a widget tree.
+pub struct Container<T: Data> {
     background: Option<PaintBrush>,
-    border: Option<BorderState>,
-}
+    border: Option<BorderStyle>,
+    corner_radius: f64,
```

Comment:
my inclination is that this should be `Option<f64>` and we should have different code paths (`Rect` vs `RoundedRect`) based on whether the option is set. But I don't have a good rationale for this, it just feels more 'correct' to me, and I'm willing to change my mind. I'd be curious for @raphlinus's opinion.

Reply:
I originally went that direction but that made the `paint` function twice as long with mostly duplicate code. It might be worth it if `Rect` is way faster than `RoundedRect` though.

Reply:
One way to solve this is to have the implementation of `RoundedRect` in piet go to the rect case when the radius is 0. We don't *currently* have that implemented, but I don't see why not, so I'm fine with simpler code.

Reply:
Sounds great. I can even look at Piet this weekend to see if I can figure out how to do that.